src/dc_mimefactory.rs

```diff
@@ -1094,8 +1096,13 @@ unsafe fn get_subject(
 ) -> *mut libc::c_char {
     let context = (*chat).context;
     let ret: *mut libc::c_char;
-    let raw_subject =
-        dc_msg_get_summarytext_by_raw((*msg).type_0, (*msg).text, &mut (*msg).param, 32, context);
+
+    let msgtext_ptr = if let Some(ref text) = (*msg).text { text.strdup() } else { ptr::null_mut() };
+
+    let raw_subject = dc_msg_get_summarytext_by_raw((*msg).type_0, msgtext_ptr, &mut (*msg).param, 32, context);
+
+    free(msgtext_ptr.cast());
```

Comment:
Maybe manage memory with CString here, like this:
```
let msgtext_c = (*msg).text.map_or(None, |text| {Some(CString::new(&text)})
let msgtext_ptr = msgtext_c.map_or(ptr::null_mut(), |text| { text.as_ptr() })
```
so you can avoid using unsafe C allocator.
Have not checked the syntax, just an idea.